### PR TITLE
Ruff: A, Avoid builtin function and type shadowing.

### PR DESCRIPTION
**Pull request by ahms5**
### Which issue(s) are closed by this pull request?

See #647

### Changes proposed in this pull request:

- apply for internal use
- how ro proceed with audio_subtypes and default_audio_subtype?

### CODE REVIEW

File: pyfar/io/io.py [hunk 1]
```diff
@@ -501,12 +502,16 @@ def audio_formats():
     return soundfile.available_formats()
 
 
-def audio_subtypes(format=None):
+@pf._utils.rename_arg(
+        {"format" : "audio_format"},
+        "'format' will be deprecated in "
+        "pyfar 0.9.0 in favor of 'audio_format'")
+def audio_subtypes(audio_format=None):
```
Comment (f-brinkmann) [inline pyfar/io/io.py:509]: I'm on favour of changing this, but we should also a test if this raises an error for pyfar>=0.9.0. There should be examples in test_deprecations for other cases.

File: pyfar/io/io.py [hunk 2]
```diff
@@ -526,12 +531,21 @@ def audio_subtypes(format=None):
...
-    return soundfile.available_subtypes(format=format)
+    return soundfile.available_subtypes(format=audio_format)
 
 
-def default_audio_subtype(format):
+@pf._utils.rename_arg(
+        {"format" : "audio_format"},
+        "'format' will be deprecated in "
+        "pyfar 0.9.0 in favor of 'audio_format'")
+def default_audio_subtype(audio_format):
```
Comment (f-brinkmann) [inline pyfar/io/io.py:541]: See above

**Review by f-brinkmann — CHANGES_REQUESTED** [review]
Thanks for the effort. I think we should add two little tests for the deprectations

**Comment by ahms5** [comment]
this should release in v0.7.0 due to the deprecation stuff

File: tests/test_deprecations.py [hunk 3]
```diff
@@ -382,4 +382,13 @@ def test_deprecations_audio_io():
     with pytest.warns(
         PyfarDeprecationWarning, match="'format' will be deprecated in "
         "pyfar 0.9.0 in favor of 'audio_format'"):
-        pf.io.audio_subtypes(format='wav')
+        pf.io.default_audio_subtype(format='wav')
+
+    if version.parse(pf.__version__) >= version.parse('0.9.0'):
+        with pytest.raises(TypeError):
+            pfilt.default_audio_subtype(None, 250, "lower", -60, None, 4,
+                                     sampling_rate=44100)
```
Comment (f-brinkmann) [inline tests/test_deprecations.py:390]:
```suggestion
            pf.io.default_audio_subtype(format='wav')
```

**Review by f-brinkmann — CHANGES_REQUESTED** [review]
I think I spotted a typo...

**Review by f-brinkmann — APPROVED** [review]
Damn you autocomplete ;)

**Review by sikersten — DISMISSED** [review]

**Review by hoyer-a — DISMISSED** [review]

**Pull request merged**